src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -215,13 +237,25 @@ private void InitializeInProcDataCollectors(string runSettings)
 
         /// <summary>
         /// Gets codebase for inproc datacollector
-        /// Uses default codebase if given path is not absolute path of inproc datacollector
+        /// Uses all codebasePaths to check where the datacollector exists
         /// </summary>
-        /// <param name="codeBase">The run Settings.</param>
+        /// <param name="codeBase">The codebase.</param>
         /// <returns> Codebase </returns>
         private string GetCodebase(string codeBase)
         {
-            return Path.IsPathRooted(codeBase) ? codeBase : Path.Combine(this.defaultCodeBase, codeBase);
+            if (!Path.IsPathRooted(codeBase))
```

Comment:
suggestion: you can flip the condition. if (path.ispathrooted(,....)) { return codebase }